### PR TITLE
Add concentration, radius, Delta_Sigma functions to halos module

### PR DESCRIPTION
**Pull request by sutieng**
## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/main/CONTRIBUTING.rst)
- [x] Write unit tests
- [x] Write documentation strings
- [ ] Assign someone from your working team to review this pull request
- [ ] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

**Comment by ntessore** [comment]
Hi Sutieng, thank you for your pull request. Could you please set up your git username and email correctly for GitHub? [There is a description in the GitHub documentation.](https://docs.github.com/en/github/getting-started-with-github/getting-started-with-git/setting-your-username-in-git)

Afterwards, you will need to fix the existing commits in this PR. The easiest that I can think of is resetting your commits without undoing any of the changes, then simply committing again, which will use the new username and email:

```console
$ git reset --soft module/halos
$ git commit
$ git push --force
```

File: skypy/halos/_colossus.py [hunk 1]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+def concentration(mass, mdef, redshift, model, cosmology, sigma8, ns):
+    r'''Halo concentration calculator.
+
+    This function calculates halo concentration(s) using the model of c-M relation
+    available in colossus.
+
+    Parameters
+    ----------
+    mass : float or array_like
+        Spherical overdensity halo mass in units of solar mass/h, corresponding
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:217]: We had a debate about the use of `h` in units. We finally decided to work in units without `h`. 
Is this relevant for you here?

File: skypy/halos/_colossus.py [hunk 2]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+        to the mass definition, mdef.
+    mdef : str
+        Halo mass definition for spherical overdensities used by colossus.
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:220]: Which are the options to choose from? Is it worth adding them here?

File: skypy/halos/_colossus.py [hunk 3]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+    sigma8 : float
+        Cosmology parameter, amplitude of the (linear) power spectrum on the
+        scale of 8 Mpc/h.
+    ns : float
+        Cosmology parameter, spectral index of scalar perturbation power spectrum.
+
+    Returns
+    -------
+    concentration : float or array_like
+        Halo concentration(s); has the same dimensions as mass.
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:236]: Units?

File: skypy/halos/_colossus.py [hunk 4]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+    fromAstropy(cosmology, sigma8=sigma8, ns=ns)
+
+    c = concentration.concentration(mass, mdef, redshift, model)
+
+    return c
+
+
+def radius(mass, concentration, redshift, mdef, Delta, cosmology, sigma8, ns):
+    r'''Calculate the scale radius and the spherical overdensity radius of halo by assuming
+    the NFW model.
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:251]:
```suggestion
    r'''Halo radii with the NFW model.
```

File: skypy/halos/_colossus.py [hunk 5]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+    from colossus.halo import concentration
+
+    fromAstropy(cosmology, sigma8=sigma8, ns=ns)
+
+    c = concentration.concentration(mass, mdef, redshift, model)
+
+    return c
+
+
+def radius(mass, concentration, redshift, mdef, Delta, cosmology, sigma8, ns):
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:249]: Is this function name descriptive enough? what about `radii_NFW` or anything better??

File: skypy/halos/_colossus.py [hunk 6]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+    r'''Calculate the scale radius and the spherical overdensity radius of halo by assuming
+    the NFW model.
+
+    This function calculates the scale radius and any spherical overdensity radius
+    for NFW dark matter halo.
+
+    Parameters
+    ----------
+    mass : float
+        A spherical overdensity halo mass in units of solar mass/h, corresponding
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:259]: See above re: discussion about units.

File: skypy/halos/_colossus.py [hunk 7]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+        to the mass definition, mdef.
+    concentration : float
+        The concentration corresponding to the given halo mass and mass definition.
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:262]: Units?

File: skypy/halos/_colossus.py [hunk 8]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+    sigma8 : float
+        Cosmology parameter, amplitude of the (linear) power spectrum on the
+        scale of 8 Mpc/h.
+    ns : float
+        Cosmology parameter, spectral index of scalar perturbation power spectrum.
+
+    Returns
+    -------
+    rs : float
+        The scale radius in physical kpc/h.
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:280]: See above re: discussion about units and `h`.

File: skypy/halos/_colossus.py [hunk 9]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+    RDelta : float
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:281]: units?

File: skypy/halos/_colossus.py [hunk 10]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+    RDelta = prof.RDelta(redshift, Delta)
+
+    return rs, RDelta
+
+
+def Delta_Sigma(mass, concentration, redshift, mdef, radius, cosmology, sigma8, ns):
+    r'''The excess surface density at given radius by assuming the NFW model.
+
+    This function uses Colossus routines to compute the excess surface density profile,
+    which is defined as Delta_Sigma(R) = Sigma(<R) − Sigma(R).
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:301]:
```suggestion
    r'''Excess surface density 
    
    This function computes the excess surface density at a given radius, assuming the NFW model.
    This function uses Colossus routines to compute the excess surface density profile,
    which is defined as Delta_Sigma(R) = Sigma(<R) − Sigma(R).
```
Or something along those lines.

File: skypy/halos/_colossus.py [hunk 11]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+
+    Parameters
+    ----------
+    mass : float
+        A spherical overdensity halo mass in units of solar mass/h, corresponding
+        to the mass definition, mdef.
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:307]: See above re: discussion about units and `h`.

File: skypy/halos/_colossus.py [hunk 12]
```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
...
+    concentration : float
+        The concentration corresponding to the given halo mass and mass definition.
+    redshift : float
+        Halo redshift
+    mdef : str
+        Halo mass definition for spherical overdensities used by colossus.
+    radius : float or array_like
+        Radius in physical kpc/h.
```
Comment (Lucia-Fonseca) [inline skypy/halos/_colossus.py:315]: See above re: discussion about units and `h`.

File: skypy/halos/tests/test_colossus.py [hunk 13]
```diff
@@ -105,3 +105,53 @@ def test_colossus_mf():
...
     assert np.all(m_halo >= m_min)
     assert np.all(m_halo <= m_max)
+
+
+@pytest.mark.skipif(not HAS_COLOSSUS, reason='test requires colossus')
+@pytest.mark.flaky
+def test_colossus_concentration():
+    from skypy.halos._colossus import concentration
+    from astropy.cosmology import Planck15
+
```
Comment (Lucia-Fonseca) [inline skypy/halos/tests/test_colossus.py:115]: We could add a comment here, briefly explaining what this test is doing.
For example:

```python
    # Check concentration array matches mass length
```
or similar

File: skypy/halos/tests/test_colossus.py [hunk 14]
```diff
@@ -105,3 +105,53 @@ def test_colossus_mf():
...
+    assert len(c) == len(mass)
+
+
+@pytest.mark.skipif(not HAS_COLOSSUS, reason='test requires colossus')
+@pytest.mark.flaky
+def test_colossus_radius():
+    from skypy.halos._colossus import radius
+    from astropy.cosmology import Planck15
+
```
Comment (Lucia-Fonseca) [inline skypy/halos/tests/test_colossus.py:131]: Add a short comment explaining.

File: skypy/halos/tests/test_colossus.py [hunk 15]
```diff
@@ -105,3 +105,53 @@ def test_colossus_mf():
...
+    assert rs > 0
+    assert RDelta > 0
+
+
+@pytest.mark.skipif(not HAS_COLOSSUS, reason='test requires colossus')
+@pytest.mark.flaky
+def test_colossus_Delta_Sigma():
+    from skypy.halos._colossus import Delta_Sigma
+    from astropy.cosmology import Planck15
+
```
Comment (Lucia-Fonseca) [inline skypy/halos/tests/test_colossus.py:148]: comment

File: skypy/halos/tests/test_colossus.py [hunk 16]
```diff
@@ -105,3 +105,53 @@ def test_colossus_mf():
...
+    mass, c, redshift = 1e+15, 3, 0.5
+    mdef = '200c'
+    cosmo = Planck15
+    sigma8, ns = 0.82, 0.96
+    radius = np.logspace(np.log10(300), np.log10(3000), 10)
+
+    DeltaSigma = Delta_Sigma(mass, c, redshift, mdef, radius, cosmo, sigma8, ns)
+
+    assert len(DeltaSigma) == len(radius)
```
Comment (Lucia-Fonseca) [inline skypy/halos/tests/test_colossus.py:157]: In general, is there anything else we can test here? Corner cases? Does any of this follow a formula for certain parameters...?

**Review by Lucia-Fonseca — CHANGES_REQUESTED** [review]
Good job!
Very nice and clear.
A few comments on:
* Units with `h` (to check)
* Docstrings
* More tests?
